src/D2L.CodeStyle.Analyzers/ApiUsage/ServiceLocator/SingletonLocatorAnalyzer.cs

```diff
@@ -32,15 +30,26 @@ public override ImmutableArray<DiagnosticDescriptor> SupportedDiagnostics
 				return;
 			}
 
-			IDictionary<INamedTypeSymbol, int> containerTypes = new [] {
-					new { TypeName = "D2L.LP.Extensibility.Activation.Domain.IPlugins`1", ContainedTypeIdx = 0 },
-					new { TypeName = "D2L.LP.Extensibility.Activation.Domain.IPlugins`2", ContainedTypeIdx = 1 },
-					new { TypeName = "D2L.LP.Extensibility.Plugins.IInstancePlugins`1",   ContainedTypeIdx = 0 },
-					new { TypeName = "D2L.LP.Extensibility.Plugins.IInstancePlugins`2",   ContainedTypeIdx = 0 }
+			var containerTypeMappings = new (string typeName, int containedTypeIdx)[] {
+					new ( "D2L.LP.Extensibility.Activation.Domain.IPlugins`1", 0 ),
+					new ( "D2L.LP.Extensibility.Activation.Domain.IPlugins`2", 1 ),
+					new ( "D2L.LP.Extensibility.Plugins.IInstancePlugins`1",   0 ),
+					new ( "D2L.LP.Extensibility.Plugins.IInstancePlugins`2",   0 )
+				};
+
+			var containerTypesBuilder = ImmutableDictionary.CreateBuilder<INamedTypeSymbol, int>( SymbolEqualityComparer.Default );
+
+			foreach( (string typeName, int containedTypeIdx) in containerTypeMappings ) {
+
+				INamedTypeSymbol? type = context.Compilation.GetTypeByMetadataName( typeName );
+				if( type.IsNullOrErrorType() ) {
+					return;
 				}
-				.Select( x => new { Type = context.Compilation.GetTypeByMetadataName( x.TypeName ), x.ContainedTypeIdx } )
-				.Where( x => !x.Type.IsNullOrErrorType() )
-				.ToDictionary( x => x.Type, x => x.ContainedTypeIdx );

```

Comment:
Wanted a symbol equality comparer, but with the anonymous type was difficult to specify the TKey, given that `SymbolEqualityComparer` is an `IEqualityComparer<ISymbol>` and the key is `INamedTypeSymbol`